tests/functional/api/groups/members_test.py

```diff
@@ -62,179 +62,135 @@ def test_it_returns_empty_list_if_no_members_in_group(self, app):
 
-        assert res.status_code == 404
-
-    def test_it_returns_404_if_malformed_userid(
-        self, app, factories, auth_client, db_session
+    def test_it_ignores_forward_users(
+        self, factories, authclient, headers, do_request, user, group
     ):
-        group = factories.Group()
-        db_session.commit()
+        forwarded_user = factories.User(authority=authclient.authority)
+        headers["X-Forwarded-User"] = forwarded_user.userid
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=group.pubid, userid="foo@bar.com"
-            ),
-            headers=self.authorization_header(auth_client),
-            expect_errors=True,
-        )
+        do_request()
 
-        assert res.status_code == 404
+        # It has added the user from the URL to the group.
+        assert user in group.members
+        # It has *not* added the user from the X-Forwarded-User header to the group.
+        assert forwarded_user not in group.members
 
-    def test_it_returns_404_if_authority_mismatch_on_group(
-        self, app, factories, user, auth_client, db_session
+    @pytest.mark.xfail
+    def test_me_alias_with_forwarded_user(
+        self, factories, authclient, do_request, headers, group
     ):
-        group = factories.Group(authority="somewhere-else.org")
-        db_session.commit()
+        forwarded_user = factories.User(authority=authclient.authority)
+        headers["X-Forwarded-User"] = forwarded_user.userid
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=group.pubid, userid=user.userid
-            ),
-            headers=self.authorization_header(auth_client),
-            expect_errors=True,
-        )
+        do_request(userid="me")
 
-        assert res.status_code == 404
+        # If the "me" alias is used with an X-Forwarded-User header it should
+        # add the forwarded user to the group.
+        assert forwarded_user in group.members
 
-    def test_it_returns_404_if_missing_auth(self, app, user, db_session, factories):
-        group = factories.Group()
-        db_session.commit()
+    def test_me_alias_with_forwarded_user_with_wrong_authority(
+        self, factories, headers, do_request
+    ):
+        forwarded_user = factories.User(authority="other")
+        headers["X-Forwarded-User"] = forwarded_user.userid
 
-        res = app.post_json(
-            "/api/groups/{pubid}/members/{userid}".format(
-                pubid=group.pubid, userid=user.userid
-            ),
-            expect_errors=True,
-        )
+        do_request(userid="me", status=404)
 
-        assert res.status_code == 404
+    def test_me_alias_without_forwarded_user(self, do_request):
+        do_request(userid="me", status=404)
 
-    def test_it_returns_404_with_token_auth(
-        self, app, token_auth_header, user, db_session, factories
-    ):
-        group = factories.Group()
```

Comment:
`group` and `user` have wrong authority so test is not testing what it intends to.